src/ol/geom/Circle.js

```diff
@@ -172,10 +172,12 @@ class Circle extends SimpleGeometry {
    */
   setCenter(center) {
     const stride = this.stride;
-    const radius = this.flatCoordinates[stride] - this.flatCoordinates[0];
+    const radiusX = this.flatCoordinates[stride] - this.flatCoordinates[0];
+    const radiusY = this.flatCoordinates[stride + 1] - this.flatCoordinates[1];
```

Comment:
y should be the same for the center and the point right of the center on the perimeter. So if we have different y values here, it's a bug that needs to be fixed before making any other changes.

Reply:
It will not be the same if the geometry has been transformed between projections where the x axes are not parallel.

Reply:
Then that's the bug that needs to be fixed.

Reply:
The API https://openlayers.org/en/latest/apidoc/module-ol_geom_Circle-Circle.html#transform explains

> This `transform` function just transforms these two points.

We need to handle that internally, but when I suggested we should avoid returning coordinates in that format to the user you did not consider it important.

Reply:
The more changes you're trying to get in here, the more I see how deep the rabbit hole goes. And fixing the transform, which is an easy fix, will make a ton of other things easier in return.

> This transform function just transforms these two points.

is cited out of context, and maybe not well expressed. It should say "This transform function just transforms the center and the radius of the circle". The overall meaning is that a circle is not a polygon with the shape of a circle, but an artificial construct with just a center and a radius.

Reply:
> Transforming a `Circle` from one coordinate system to another means: 1. transform the center (first coordinate), 2. convert the radius from source crs units to destination crs units, 3. encode the radius in the second coordinate.

It is not as simple as that as the reference points will be on different alignments, so extrapolation may be needed as in https://github.com/openlayers/openlayers/pull/14672/files#diff-90b8ad0e1e436e034c530c6de89696519c2f2308e8b90d9795325599f34a8a4fR294 to produce a visually similar result.